src/worker.ts

```diff
@@ -0,0 +1,75 @@
+import { Value } from "@sinclair/typebox/value";
+import { run } from "./run";
+import { Env, envConfigValidator } from "./types/env";
+import { assistivePricingSettingsSchema } from "./types/plugin-input";
+
+export default {
+  async fetch(request: Request, env: Env): Promise<Response> {
+    try {
+      if (request.method !== "POST") {
+        return new Response(JSON.stringify({ error: `Only POST requests are supported.` }), {
+          status: 405,
+          headers: { "content-type": "application/json", Allow: "POST" },
+        });
+      }
+      const contentType = request.headers.get("content-type");
+      if (contentType !== "application/json") {
+        return new Response(JSON.stringify({ error: `Error: ${contentType} is not a valid content type` }), {
+          status: 400,
+          headers: { "content-type": "application/json" },
+        });
+      }
+      if (!envConfigValidator.test(env)) {
+        const errorDetails: string[] = [];
+        for (const error of envConfigValidator.errors(env)) {
+          errorDetails.push(`${error.path}: ${error.message}`);
+        }
+        return new Response(JSON.stringify({ error: `The environment is invalid: ${errorDetails.join("; ")}` }), {
```

Comment:
This sounds like an error. Should be 5xx

We can also consider custom error codes like how cloudflare does it 1xxx

Reply:
This is a payload error so usually referred as `4XX`. `5XX` means the server crashed handling the request, which is not the case here, just gracefully denied the request.